packages/web-components/src/components/footer/language-selector.ts

```diff
@@ -0,0 +1,153 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2020, 2021
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { html, property, customElement, query } from 'lit-element';
+import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings.js';
+import BXComboBox, { DROPDOWN_SIZE } from 'carbon-web-components/es/components/combo-box/combo-box.js';
+import BXComboBoxItem from 'carbon-web-components/es/components/combo-box/combo-box-item.js';
+import HostListener from 'carbon-web-components/es/globals/decorators/host-listener.js';
+import HostListenerMixin from 'carbon-web-components/es/globals/mixins/host-listener.js';
+import { FOOTER_SIZE } from './footer';
+import styles from './footer.scss';
+import { findIndex, forEach } from '../../globals/internal/collection-helpers';
+
+const { stablePrefix: ddsPrefix } = ddsSettings;
+
+/**
+ * Language selector component.
+ * The API for language selection is still subject to change.
+ *
+ * @element dds-language-selector
+ * @internal
+ */
+@customElement(`${ddsPrefix}-language-selector`)
+class DDSLanguageSelector extends HostListenerMixin(BXComboBox) {
+  /**
+   * Property that saves the last valid language to use on reset cases.
+   * @private
+   */
+
+  @property({ attribute: 'last-valid-lang' })
+  lastValidLang;
```

Comment:
See other code in our codebase using `@internalProperty` as well as https://lit-element.polymer-project.org/guide/properties#declare-with-decorators for more details. Also please make sure the code referring to `_lastValidLang` is updated to add `_` prefix:

```suggestion
  /**
   * Property that saves the last valid language to use on reset cases.
   */
  @internalProperty()
  private _lastValidLang?: string;
```